### DISCUSSION TITLE
Compile Failure

### DISCUSSION THREAD

**Discussion by dizhizhou** [trailofbits/vast#527] [Q&A]
Hi community,

VAST compile fails in my local ubuntu due to the following link errors:

Commands:
`vast$ cmake --preset ninja-multi-default     --toolchain ./cmake/lld.toolchain.cmake     -DCMAKE_PREFIX_PATH=/usr/local/llvm-17/ -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DLLVM_ENABLE_LLD=ON -DLLVM_ENABLE_RTTI=ON

// There are many of these link errors. I only list one of them
vast$ cmake --build --preset ninja-rel
.....
[188/191] Linking CXX executable tools/vast-lsp-server/Release/vast-lsp-server
FAILED: tools/vast-lsp-server/Release/vast-lsp-server
.......
ld.lld: error: undefined symbol: typeinfo for mlir::Dialect
>>> referenced by UnsupportedDialect.cpp
>>>               UnsupportedDialect.cpp.o:(typeinfo for vast::unsup::UnsupportedDialect) in archive Release/lib/libVASTUnsupported.a
>>> referenced by MetaDialect.cpp
>>>               MetaDialect.cpp.o:(typeinfo for vast::meta::MetaDialect) in archive Release/lib/libVASTMeta.a
>>> referenced by LowLevelDialect.cpp
>>>               LowLevelDialect.cpp.o:(typeinfo for vast::ll::LowLevelDialect) in archive Release/lib/libVASTLowLevel.a
>>> referenced 3 more times
>>> did you mean: vtable for mlir::Dialect
>>> defined in: /usr/local/llvm-17/lib/libMLIRIR.a(Dialect.cpp.o)
`

It seems that MLIR builds are not linked correctly.

LLVM&MLIR project(I build llvm-17 from source code) is  built correclty and the install path is /usr/local/llvm-17/ which is already in the vast cmake config CLI

```
/mnt/data/vast$ find /usr/local/llvm-17/ -name LLVMConfig.cmake
/usr/local/llvm-17/lib/cmake/llvm/LLVMConfig.cmake
/mnt/data/vast$ find /usr/local/llvm-17/ -name MLIRConfig.cmake
/usr/local/llvm-17/lib/cmake/mlir/MLIRConfig.cmake
```

Thoughts?

**Comment by dizhizhou** [comment]
Solved. It turns out that my LLVM was compiled without DLLVM_ENABLE_RTTI=ON

**Comment by lkorenc** [comment]
Glad you were able to solve it! If you stumble upon any other issues do not hesitate to ask.
